src/SDKs/Azure.Core/data-plane/Azure.Core/Pipeline/Policies/TaskExtensions.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Diagnostics;
+using System.Threading.Tasks;
+
+namespace Azure.Core.Pipeline.Policies
+{
+    internal static class TaskExtensions
+    {
+        public static T AssertCompleted<T>(this Task<T> task)
```

Comment:
Should we consider naming this something like `EnsureCompleted` to avoid confusion with test semantics?   I know that its calling `Debug.Assert` but this could be mixed up with a helper for test assertions at a glance.